### PR TITLE
WIP: Sentinel support

### LINKED ISSUE #146: Hope to support redis sentinel mode
hi, dear author. Hope to support redis sentinel mode, or you can integrate predixy into the operator as a proxy, that might be better

### PR DESCRIPTION
**Pull request by davidcollom**
Support for a Redis Sentinel Configuration, where users don't need/want a RedisCluster but need HA.

Closes #146

### CODE REVIEW

**Comment by samof76** [comment]
@davidcollom  is this feature ready?

File: config/crd/bases/redis.redis.opstreelabs.in_redisclusters.yaml [hunk 1]
```diff
@@ -274,8 +274,7 @@ spec:
                 - image
                 type: object
               redisFollower:
-                description: RedisFollower interface will have the redis follower
-                  configuration
+                description: RedisLeader interface will have the redis leader configuration
```
Comment (samof76) [inline config/crd/bases/redis.redis.opstreelabs.in_redisclusters.yaml:277]: This description is wrong.

**Comment by davidcollom** [comment]
@samof76 It is not - sorry, my workload has since increased in that this is still very much WiP - If there is sufficient appetite for this feature I'm happy to continue development once my availability drops.

**Comment by samof76** [comment]
@davidcollom sure there is a lot stuff that still require sentinel to be operational :)

**Comment by shubham-cmyk** [comment]
#408

**Pull request closed without merging**